data/lua/wml/objectives.lua

```diff
@@ -87,7 +87,7 @@ local function generate_objectives(cfg)
 					.. color_prefix(r, g, b) .. objective_bullet.. description
 					.. turn_counter .. "</span>" .. "\n"
 			else
-				wml.error("[objective] Either condition= is missing or condition= has invalid value (ignoring).")
+				wml.error(stringx.vformat("[objective] Unknown condition '$condition', ignoring.", { condition = condition }))
```

Comment:
If the condition is missing then this isn't going to be very understandable; maybe have an `elseif condition == "" then` with a specific message for that.

Reply:
You really think it's difficult to understand that `[objective] Unknown condition '', ignoring.` means the condition is emtpy?

Reply:
It's pretty understandable to me. I'm not sure if it would be understandable to everyone.

Reply:
I won't block merging this, but still feel that adding the extra message would be worthwhile, otherwise it's a message talking about a `condition` in a block of WML that doesn't mention a `condition` at all. It'll be obvious to the UMC author once they've found the place that needs changing, but it's giving them a misleading search term for finding that place.

Reply:
Ah, you're talking about `condition == nil`.

Reply:
Added a message above for the nil case.